sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -108,4 +108,10 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
       }
     }
   }
+
+  override def getScanExecPartitionSize(query: String): Long = {
+    sql(query).queryExecution.sparkPlan.collectFirst {
+      case p: FileSourceScanExec => p
+    }.get.relation.location.inputFiles.length
```

Comment:
this is not partitions number, but files number.

This should be `...relation.selectedPartitions.length`

Reply:
> this is not partitions number, but files number.
> 
> This should be `...relation.selectedPartitions.length`

make FileSourceScanExec's selectedPartitions to be public like HiveTableScanExec's prunedPartitions will be ok